src/Transform/ONNX/ConstProp.td

```diff
@@ -185,6 +187,25 @@ def SqrtofConst :  Pat<
 // Exactly the same pattern as for the elementwise ADD operations.
 //===----------------------------------------------------------------------===//
 
+def ReduceZeroAdd : Pat<
+  // From add(x, 0).
+  (ONNXAddOp $x,(ONNXConstantOp:$c1 $s1, $v1)),
+  // To x
+  (replaceWithValue $x),
+  [(isDenseZeroTensor $v1)]>;
+
+//something off about the shapes
+def MulAddDistribitive : Pat<
```

Comment:
When do you know the transformed version is better than the original?
Assuming ONNXMulOp of the two constants is done at compile time, both are 2 ops. One does add followed by mul, the other does vice versa.
Does ONNX MLIR figure out which of the two cases can do more constant propagation?

Reply:
The Mul of the 2 constants does happen at compile time. In general this reduces a chain of (C(A(x+a)+ b)+c) ... into a operation like Ax+b. I don't think that chain is currently reduced by any of the other other ONNX Const Propagations. The current ONNX const propagations optimize away associative additions and this change does not affect that. If anything it can only create more associative additions (like a(x+b) + c(y+d) = ax + cy + (ab+cd))

Reply:
Ah I see. Makes sense.